src/SpotifySongSearch.jsx

```diff
@@ -40,46 +43,122 @@ const SpotifySongSearch = ({ authorized }) => {
   };
 
   useEffect(() => {
-    if (!authorized || !userData) return;
+    if (!userData || !authorized) return;
 
     const text = getTracks();
     setRawText(text);
     performRequest(text, userData.token);
   }, [authorized, userData]);
 
-  if (!authorized) return <h1>Waiting for authorization</h1>;
-
   return (
-    <div className="searchContainer">
-      <p>Enter track list (in &quot;artist&quot; - &quot;title&quot; format)</p>
-      <textarea
-        className="searchBox"
-        onChange={handleTracksChange}
-        value={rawText}
-      />
-      <button className="searchButton" type="button" onClick={handleClick}>
-        Search
-      </button>
-      {isFetching ? (
-        <p>Fetching...</p>
-      ) : (
-        <ResponsesView responses={responses} />
-      )}
-      <CreatePlaylistButton
-        tracksUris={responses
-          .filter((response) => !response.hasError)
-          .map((track) => track.trackUri)}
+    <>
+      <SearchForm
+        handleClick={handleClick}
+        handleTracksChange={handleTracksChange}
+        rawText={rawText}
       />
+      <ResponsesContainer isFetching={isFetching} responses={responses} />
+      <CreatePlaylistForm responses={responses} />
+    </>
+  );
+};
+
+const ResponseType = PropTypes.shape({
+  trackUri: PropTypes.string,
+  albumCover: PropTypes.string,
+  artists: PropTypes.arrayOf(PropTypes.string),
+  album: PropTypes.string,
+  name: PropTypes.string,
+  duration: PropTypes.number,
+  hasError: PropTypes.bool,
+});
+
+const CreatePlaylistForm = ({ responses }) => {
```

Comment:
this file is growing, so make sense to put separate components into separate files